### PR TITLE
user/vpidatala/19574990 - adding DCR templates

### PR DESCRIPTION
**Pull request by vpidatala94**
added new VMI DCR Templates Back to Repo here so that we can raise pull requests in the future.  Further the change I am introducing the change to the ARM Templates is to introduce a parameter for DcrName to let Customers create a custom DCR also see the default value if the customer doesn't wish to provide the name.

### CODE REVIEW

File: OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        "WorkspaceLocation": {
+            "value": "<WorkspaceLocation>"
+        },
+        "userGivenDcrName": {
+            "type": "String",
+            "metadata": {
+              "displayName": "Name of the Data Collection Rule(DCR)",
+              "description": "This is the name of the AMA-VMI Data Collection Rule(DCR)"
+            },
+            "defaultValue": "ama-vmi-default-dcr"
```
Comment (jatakiajanvi12) [inline OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json]: May be add the perf and map indication in the DCR name. No need of default keyword.

File: OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        "WorkspaceResourceId": {
+            "value": "/subscriptions/<SubscriptionId>/resourceGroups/<ResourceGroup>/providers/Microsoft.OperationalInsights/workspaces/<WorkspaceName>"
+        },
+        "WorkspaceLocation": {
+            "value": "<WorkspaceLocation>"
+        },
+        "userGivenDcrName": {
+            "type": "String",
+            "metadata": {
+              "displayName": "Name of the Data Collection Rule(DCR)",
```
Comment (jatakiajanvi12) [inline OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json]: Do we need the metadata here?

File: OnbordingTemplates/DeployDcr/PerfOnlyDcr/DeployDcrParameters.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        "WorkspaceLocation": {
+            "value": "<WorkspaceLocation>"
+        },
+        "userGivenDcrName": {
+            "type": "String",
+            "metadata": {
+              "displayName": "Name of the Data Collection Rule(DCR)",
+              "description": "This is the name of the AMA-VMI Data Collection Rule(DCR)"
+            },
+            "defaultValue": "ama-vmi-default-dcr"
```
Comment (jatakiajanvi12) [inline OnbordingTemplates/DeployDcr/PerfOnlyDcr/DeployDcrParameters.json]: Same comments as above

File: OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrTemplate.json [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "parameters": {},
+                    "variables": {},
+                    "resources": [
+                        {
+                            "type": "Microsoft.Insights/dataCollectionRules",
+                            "apiVersion": "2021-04-01",
```
Comment (jatakiajanvi12) [inline OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrTemplate.json:42]: Is this the latest api version? Do we need to update the api?

File: OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json (hunk 1 above)
Comment (vpidatala94) [reply OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json]: sure, will make the, change, default name is line with the Ux offering where a default name is provided, this is based on previous discussion oncall about this.

File: OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json (hunk 2 above)
Comment (vpidatala94) [reply OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json]: This is legacy code, I only added a code to add parameter to ask customers to provide name to DCR, I think the metadata is helpful for the developers to know what the policy does.

File: OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrTemplate.json (hunk 4 above)
Comment (vpidatala94) [reply OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrTemplate.json:42]: Wil review that, thank you for your comments.

File: OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json (hunk 2 above)
Comment (jatakiajanvi12) [reply OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrParameters.json]: It should be there in the template but not the parameters file

File: OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrTemplate.json [hunk 5]
```diff
@@ -20,14 +20,14 @@
               "displayName": "Name of the Data Collection Rule(DCR)",
               "description": "This is the name of the AMA-VMI Data Collection Rule(DCR)"
             },
-            "defaultValue": "ama-vmi-default-dcr"
+            "defaultValue": "ama-vmi-default-perfAndda-dcr"
```
Comment (jatakiajanvi12) [inline OnbordingTemplates/DeployDcr/PerfAndMapDcr/DeployDcrTemplate.json:23]: nit: If going for camelcase naming - perfAndDa?

**Review by jatakiajanvi12 — APPROVED** [review]

**Pull request merged**